packages/api/src/Api.ts

```diff
@@ -26,14 +26,27 @@ import getPlugins from './plugins';
 import staticMetadata from './staticMetadata';
 import {ApiOptions, IPlugin} from './types';
 import {getProvider} from './util/getProvider';
+import {getTimeout} from './util/getTimeout';
 import {injectOption, injectPlugins, mergePlugins} from './util/injectPlugin';
 import logger from './util/logging';
 
 const Types = require('@cennznet/types');
 
+export const DEFAULT_TIMEOUT = 10000;
+
 export class Api extends ApiPromise {
     static async create(options: ApiOptions | ProviderInterface = {}): Promise<Api> {
-        return new Api(options).isReady;
+        const api = await new Api(options);
+        return withTimeout(
+            new Promise((resolve, reject) => {
+                api.isReady.then(resolve, reject);
+
+                api.once('error', err => {
```

Comment:
remove all listeners from eventemitter and disconnect provider when connection fails